src/Toolkit/Toolkit/Internal/TimeExtensions.cs

```diff
@@ -122,7 +122,8 @@ public static DateTimeOffset AddTimeValue(this DateTimeOffset startTime, TimeVal
         /// <param name="timeExtent">The extent to divide.</param>
         /// <param name="count">The amount to divide the extent by.</param>
         /// <returns>A TimeValue instance which, will fit evenly into the input TimeExtent the specified number of times.</returns>
-        public static TimeValue Divide(this TimeExtent timeExtent, int count)
+        [return: System.Diagnostics.CodeAnalysis.NotNullIfNotNull("timeExtent")]
+        public static TimeValue? Divide(this TimeExtent? timeExtent, int count)
         {
             if (timeExtent == null)
             {
```

Comment:
Looks like `Union(...)`, below, was designed to handle a null "otherTimeExtent" value.  In fact we pass nullable properties (Layer.FullTimeExtent) as that parameter in TimeSlider.cs &mdash; I'm surprised that no warning was generated.

Reply:
A warning wasn't generated because it had a null-check. I'll remove that null check and allow null here.